crates/oxc_semantic/src/jsdoc/parser/parse.rs

```diff
@@ -35,14 +35,14 @@ impl<'a> JSDocParser<'a> {
         let mut tags = vec![];
 
         // Let's start with the first `@`
-        while let Some(c) = self.source_text.chars().nth(self.current) {
+        while let Some((_, c)) = self.source_text[self.current..].char_indices().next() {
             match c {
                 '@' => {
-                    self.current += 1;
+                    self.current += c.len_utf8();
```

Comment:
I think you can write `self.current = i` given `char_indices` gives you the character offset.

Reply:
```diff
-        while let Some((_, c)) = self.source_text[self.current..].char_indices().next() {
+        while let Some((i, c)) = self.source_text[self.current..].char_indices().next() {
             match c {
                 '@' => {
-                    self.current += c.len_utf8();
+                    self.current = i;
                     tags.push(self.parse_tag());
                 }
                 _ => {
-                    self.current += c.len_utf8();
+                    self.current = i;
                 }
             }
         }
```

I tried this, but when I did, the test hung and stopped working. What am I missing?